homeassistant/components/rachio/switch.py

```diff
@@ -55,23 +62,64 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+ATTR_PERCENT = "percent"
 ATTR_SCHEDULE_SUMMARY = "Summary"
 ATTR_SCHEDULE_ENABLED = "Enabled"
 ATTR_SCHEDULE_DURATION = "Duration"
+ATTR_SCHEDULE_TYPE = "Type"
 ATTR_ZONE_NUMBER = "Zone number"
 ATTR_ZONE_SHADE = "Shade"
 ATTR_ZONE_SLOPE = "Slope"
 ATTR_ZONE_SUMMARY = "Summary"
 ATTR_ZONE_TYPE = "Type"
 
 
+SERVICE_SET_ZONE_MOISTURE_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+        vol.Required(ATTR_PERCENT): cv.positive_int,
+    }
+)
+
+
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up the Rachio switches."""
     # Add all zones from all controllers as switches
```

Comment:
Can we use entity services here instead?

https://developers.home-assistant.io/docs/dev_101_services/#entity-services

Reply:
Just tried and that was so much less work, wish I would have thought of that!  I'll get it updated.